File: dashboard/frontend/src/types/insights.ts
export interface SecurityFinding {
  severity: string
  title: string
  description: string
  affected_resources: string[]
  recommendation: string
  confidence: number
}

export interface CostAnomalyFinding {
  resource_id: string
  resource_name: string
  service_type: string
  expected_cost: number
  actual_cost: number
  deviation_pct: number
  recommendation: string
}

export interface OpsRisk {
  severity: string
  title: string
  description: string
  affected_resources: string[]
  recommendation: string
}

export interface InsightResult {
  summary: string
  findings?: SecurityFinding[]
  anomalies?: CostAnomalyFinding[]
  risks?: OpsRisk[]
  recommendations?: string[]
  risk_score?: number
  estimated_monthly_savings_usd?: number
  optimization_candidates?: Record<string, unknown>[]
  generated_at: string
  type: string
}

export interface ComprehensiveInsight {
  security: InsightResult | null
  cost: InsightResult | null
  operational: InsightResult | null
  generated_at: string
  type: string
}

File: dashboard/backend/app/schemas/insights.py
from pydantic import BaseModel


class SecurityFinding(BaseModel):
    severity: str
    title: str
    description: str
    affected_resources: list[str]
    recommendation: str
    confidence: float


class SecurityInsight(BaseModel):
    summary: str
    findings: list[SecurityFinding]
    risk_score: float


class CostAnomalyFinding(BaseModel):
    resource_id: str
    resource_name: str
    service_type: str
    expected_cost: float
    actual_cost: float
    deviation_pct: float
    recommendation: str


class CostInsight(BaseModel):
    summary: str
    anomalies: list[CostAnomalyFinding]
    optimization_candidates: list[dict]
    estimated_monthly_savings_usd: float


class OpsRisk(BaseModel):
    severity: str
    title: str
    description: str
    affected_resources: list[str]
    recommendation: str


class OpsInsight(BaseModel):
    summary: str
    risks: list[OpsRisk]
    recommendations: list[str]


class ComprehensiveInsight(BaseModel):
    security: SecurityInsight | None = None
    cost: CostInsight | None = None
    operational: OpsInsight | None = None
